tests/storage_test.go

```diff
@@ -165,5 +170,78 @@ var _ = Describe("Storage", func() {
 				close(done)
 			}, 240)
 		})
+
+		Context("With ephemeral alpine PVC", func() {
+			// The following case is mostly similar to the alpine PVC test above, except using different VM.
+			It("should be successfully started", func(done Done) {
+				checkReadiness()
+
+				// Start the VM with the PVC attached
+				vm := tests.NewRandomVMWithEphemeralPVC(tests.DiskAlpineISCSI)
+				vm.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nodeName}
+				RunVMAndExpectLaunch(vm, false, 45)
+
+				expecter, _, err := tests.NewConsoleExpecter(virtClient, vm, "serial0", 10*time.Second)
+				defer expecter.Close()
+				Expect(err).To(BeNil())
+
+				By("Checking that the VM console has expected output")
+				_, err = expecter.ExpectBatch([]expect.Batcher{
+					&expect.BExp{R: "Welcome to Alpine"},
+				}, 200*time.Second)
+				Expect(err).To(BeNil())
+
+				close(done)
+			}, 240)
+
+			It("should not persist data", func(done Done) {
```

Comment:
I fear you will have to be a little bit more creative here. Alpine and Cirros both boot read-only. That might actually be pretty tricky to solve without a storage provisioner. We could add an empy "PVC" which we clean after the tests. Not sure ...

Reply:
They boot rw (tmpfs on / type tmpfs (rw,relatime,mode=755)), the tricky part is tmpfs...

Reply:
Oh that is great, then you can just mount the disk again on a different location and write there to it.

Reply:
oh that probably does not help, since if I bindmount it it will probably also point to the tmpfs.

Reply:
And to make matters wrose, /dev/sda1 is formatted as iso9660. :/